dcp_diag/component_agents/analysis_agent.py

```diff
@@ -0,0 +1,171 @@
+                "labels": {
+                    "bundle-uuid": "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee",
+                    "bundle-version": "2018-11-02T114842.872218Z",
+                    "caas-collection-name": "xxxx-prod",
+                    "cromwell-workflow-id": "cromwell-aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee",
+                    "project_shortname": "project name",
+                    "project_uuid": "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee",
+                    "workflow-name": "AdapterSmartSeq2SingleCell",
+                    "workflow-version": "smartseq2_v2.1.0"
+                },
+                "name": "AdapterSmartSeq2SingleCell",
+                "start": "2019-01-06T20:18:18.102Z",
+                "status": "Succeeded",
+                "submission": "2019-01-06T20:16:30.804Z"
+            }
+
+        Args:
+            uuid (str): Secondary-analysis service (Cromwell) workflow UUID.
+
+        Returns:
+            Workflow: Workflow object.
+
+        Raises:
+            requests.exceptions.HTTPError: When the request to Secondary-analysis service (Cromwell) failed.
+        """
+        query_dict = {
+            'id': uuid,
+            'additionalQueryResultFields': ['labels']
+        }
+        response = cwm_api.query(query_dict=query_dict, auth=self.auth)
+        response.raise_for_status()
+        result = response.json()
+        all_workflows = result['results']
+        total_count = result['totalResultsCount']
+        assert len(all_workflows) == total_count == 1
+        return Workflow(all_workflows[0])
+
+    def query_by_bundle(self, bundle_uuid, bundle_version=None):
+        """Query the analysis workflows by their workflow-UUID.
+
+        Note, due to the open issue: https://github.com/broadinstitute/cromwell/issues/3115, if the result of workflows
+        are more than ~1000, this function will very likely raise an error.
+
+        Args:
+            bundle_uuid (str): HCA DCP bundle UUID.
+            bundle_version (str): Optional, HCA DCP bundle version. By default, it's None.
+
+        Returns:
+            Tuple[int, List[Workflow]]: A tuple of total count of workflows and the list of Workflow objects. E.g.
+                (100, [Workflow_1, ..., Workflow_100])
+
+        Raises:
+            requests.exceptions.HTTPError: When the request to Secondary-analysis service (Cromwell) failed.
+        """
+        query_dict = {
+            'label': {
+                'bundle-uuid': bundle_uuid
+            },
+            'additionalQueryResultFields': ['labels']
+        }
+
+        if bundle_version:
+            query_dict['label']['bundle-version'] = bundle_version
+
+        response = cwm_api.query(query_dict=query_dict, auth=self.auth)
+        response.raise_for_status()
+        result = response.json()
+        all_workflows = [Workflow(wf) for wf in result['results']]
+        total_count = result['totalResultsCount']
+        return total_count, all_workflows
```

Comment:
I think it's simpler to just have the function return `all_workflows` and then get the len(all_workflows) wherever you need the number of workflows. Same for `query_by_project_uuid`. What do you think?

Reply:
You are right! Will change!